### PR TITLE
Adding documentation for pre-push hooks

### PR DESCRIPTION
**Pull request by melkodary**
[Pre-Push hook for linting ](https://github.com/oakestra/oakestra/issues/278)
[PR](https://github.com/oakestra/oakestra/pull/279) of implementation

Screenshot 2024-03-20 at 16 57 38

### CODE REVIEW

File: content/docs/contribute/git-hooks-guide.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+### Why Git hooks
+
+Git pipelines often fail because of linting issues, which then requires another commit to fix it, leaving the history of commits just filled with "liniting fix".
+
+## Why `pre-commit`
+
+Git hooks are not shared across the repo, and it's only avaiable locally. This means every developer will need to implement their own hooks. Hence tool called `pre-commit`. The tool name `pre-commit` name is misleading as it implies it only provides pre-commit hooks. However it also provides other stages such as `pre-push`.
+
+Through this tool, developers define a configuration file that specifies what
```
Comment (Malyuk-A) [inline content/docs/contribute/git-hooks-guide.md:21]: I think this sentence lacks a proper ending?

File: content/docs/contribute/git-hooks-guide.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+1. First, ensure that you have pre-commit installed. It can be installed via pip or Homebrew (for macOS users):
+
+```sh
+pip install pre-commit
+# or
+brew install pre-commit
+```
+
+2. **Install the git hooks scripts**:
+In Oakestra repo, there is a file called `.pre-commit-config.yaml` containing git hooks. They can be installed by running:
```
Comment (Malyuk-A) [inline content/docs/contribute/git-hooks-guide.md:36]:
```suggestion
In the Oakestra repo, there is a file called `.pre-commit-config.yaml` containing git hooks. They can be installed by running:
```

File: content/docs/contribute/git-hooks-guide.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+Git hooks are scripts that Git executes before or after events such as: pre-commit, pre-push. These hooks are a powerful part of the Git ecosystem, enabling custom actions based on the current repository state or the changes being committed. They are used for a variety of tasks such as syntax checking, testing, linting, and auto-formatting code before it is committed or pushed. More information on git hooks is found [here](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks)
+
+### Why Git hooks
+
+Git pipelines often fail because of linting issues, which then requires another commit to fix it, leaving the history of commits just filled with "liniting fix".
+
+## Why `pre-commit`
+
+Git hooks are not shared across the repo, and it's only avaiable locally. This means every developer will need to implement their own hooks. Hence tool called `pre-commit`. The tool name `pre-commit` name is misleading as it implies it only provides pre-commit hooks. However it also provides other stages such as `pre-push`.
```
Comment (Malyuk-A) [inline content/docs/contribute/git-hooks-guide.md:19]:
```suggestion
Git hooks are not shared across the repo, and it's only available locally. This means every developer will need to implement their own hooks. Hence we use a tool called `pre-commit`. The tool name `pre-commit` name is misleading as it implies it only provides pre-commit hooks. However, it also provides other stages such as `pre-push`.
```

File: content/docs/contribute/git-hooks-guide.md (hunk 2 above)
Comment (Malyuk-A) [inline content/docs/contribute/git-hooks-guide.md:36]: I cannot find this file ?
image

Comment (melkodary) [reply content/docs/contribute/git-hooks-guide.md:36]: https://github.com/oakestra/oakestra/pull/279

PR is not yet merged.

Comment (melkodary) [reply content/docs/contribute/git-hooks-guide.md]: Waiting for this to be approved first before the merging the implementation PR.

**Comment by Malyuk-A** [comment]
Changes look good to me.

**Comment by melkodary** [comment]
Tagging you here, @nitindermohan @giobart, as I am unable to add reviewers.

**Review by giobart — APPROVED** [review]

**Review by nitindermohan — APPROVED** [review]
This looks good. Approved!

**Pull request merged**